### PR TITLE
[Feature] Refactor tests

### PR DESCRIPTION
**Pull request by brapoprod**
This PR refactors the tests to be more streamlined and consistent. Also way more safe since we test towards what is send to the EditorAPI instead of just checking if the method has been called (in a lot of cases)

## Related tickets

-   [WRS-1103](https://support.chili-publish.com/projects/WRS/issues/WRS-1103)

### CODE REVIEW

**Review by pkgacek — APPROVED** [review]

**Review by NicolaVerbeeck — APPROVED** [review]
so-so-good

File: src/tests/controllers/CharacterStyleController.test.ts [hunk 1]
```diff
@@ -76,31 +87,31 @@ const updateCharacterStyle: CharacterStyleUpdate = {
...
+        await mockedCharacterStyleController.duplicateCharacterStyle('6');
+        expect(mockEditorApi.duplicateCharacterStyle).toHaveBeenCalledTimes(1);
+        expect(mockEditorApi.duplicateCharacterStyle).toHaveBeenCalledWith('6');
 
-        mockedCharacterStyleController.renameCharacterStyle('5', 'name');
-        expect(mockedCharacterStyleController.renameCharacterStyle).toHaveBeenCalledTimes(1);
-        expect(mockedCharacterStyleController.renameCharacterStyle).toHaveBeenCalledWith('5', 'name');
+        await mockedCharacterStyleController.renameCharacterStyle('5', 'name');
+        expect(mockEditorApi.renameCharacterStyle).toHaveBeenCalledTimes(1);
+        expect(mockEditorApi.renameCharacterStyle).toHaveBeenCalledWith('5', 'name');
```
Comment (Matthiee) [inline src/tests/controllers/CharacterStyleController.test.ts:115]: All tests are grouped.

File: src/tests/controllers/ColorStyleController.test.ts [hunk 2]
```diff
@@ -1,60 +1,74 @@
...
-            r: 1,
-            g: 1,
-            b: 1,
-            colorType: ColorType.rgb,
-        });
-
-        mockedColorStyleController.removeColor('4');
-        expect(mockedColorStyleController.removeColor).toHaveBeenCalledTimes(1);
-        expect(mockedColorStyleController.removeColor).toHaveBeenCalledWith('4');
+describe('ColorStyleController', () => {
```
Comment (Matthiee) [inline src/tests/controllers/ColorStyleController.test.ts:34]: The tests are grouped.

File: src/tests/controllers/ConfigurationConnectorController.test.ts [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
-describe('ConfigurationConnector methods', () => {
+describe('ConfigurationController', () => {
     it('Should call all of the configuration functions of child successfully', async () => {
         const configKey = 'dam';
         const configValue = 'm123';
 
-        await mockedSDK.configuration.setValue(configKey, configValue);
-        expect(mockedSDK.editorAPI.setConfigValue).toHaveBeenCalledTimes(1);
-        expect(mockedSDK.editorAPI.setConfigValue).toHaveBeenCalledWith(configKey, configValue);
+        await mockedConfigurationController.setValue(configKey, configValue);
```
Comment (Matthiee) [inline src/tests/controllers/ConfigurationConnectorController.test.ts:26]: The tests are grouped.

File: src/tests/controllers/ConnectorController.test.ts [hunk 4]
```diff
@@ -35,14 +46,14 @@ describe('Connector methods', () => {
         const headerName = 'headerName';
         const headerValue = 'headerValue';
 
-        await mockedSDK.connector.getState(connectorId);
-        expect(mockedSDK.editorAPI.getConnectorState).toHaveBeenCalledTimes(1);
+        await mockedConnectorController.getState(connectorId);
```
Comment (Matthiee) [inline src/tests/controllers/ConnectorController.test.ts:49]: The tests are grouped.

File: src/tests/controllers/DebugController.test.ts [hunk 5]
```diff
@@ -1,28 +1,36 @@
...
-    it('Should call all of the Debug Functions of EditorAPI successfully', () => {
-        mockedDebugProperties.getLogs();
-        expect(mockedDebugProperties.getLogs).toHaveBeenCalledTimes(1);
-        mockedDebugProperties.toggleDebugPanel();
-        expect(mockedDebugProperties.toggleDebugPanel).toHaveBeenCalledTimes(1);
-        mockedDebugProperties.enableDebug();
-        expect(mockedDebugProperties.enableDebug).toHaveBeenCalledTimes(1);
-        mockedDebugProperties.disableDebug();
-        expect(mockedDebugProperties.disableDebug).toHaveBeenCalledTimes(1);
+    it('Should call all of the Debug Functions of EditorAPI successfully', async () => {
```
Comment (Matthiee) [inline src/tests/controllers/DebugController.test.ts:26]: The tests are grouped.

File: src/tests/controllers/FrameController.test.ts [hunk 6]
```diff
@@ -78,222 +138,228 @@ afterAll(() => {
 
 describe('FrameProperties', () => {
     it('Should call all of the Frame Functions of EditorAPI successfully', async () => {
-        await mockedSDK.frame.addFrame(FrameTypeEnum.image, 100, 100, 100, 100);
-        expect(mockedSDK.editorAPI.addFrame).toHaveBeenCalledTimes(1);
-        expect(mockedSDK.editorAPI.addFrame).toHaveBeenCalledWith(FrameTypeEnum.image, 100, 100, 100, 100);
+        await mockedFrameController.addFrame(FrameTypeEnum.image, 100, 100, 100, 100);
```
Comment (Matthiee) [inline src/tests/controllers/FrameController.test.ts:141]: The tests are grouped.

File: src/tests/controllers/LayoutController.test.ts [hunk 7]
```diff
@@ -37,65 +54,65 @@ afterEach(() => {
 
 describe('Layout methods', () => {
     it('handles all layout methods', async () => {
-        await mockedSDK.layout.getLayouts();
-        expect(mockedSDK.editorAPI.getLayouts).toHaveBeenCalledTimes(1);
+        await mockedFontController.getLayouts();
```
Comment (Matthiee) [inline src/tests/controllers/LayoutController.test.ts:57]: The tests are grouped.

File: src/tests/controllers/SubscriberContoller.test.ts [hunk 8]
```diff
@@ -1,101 +1,129 @@
...
+    jest.spyOn(mockEditorApi, 'onUndoStackStateChanged');
 });
 
 afterEach(() => {
     jest.restoreAllMocks();
 });
 describe('Subscriber methods', () => {
     it('Should call all of the subscriber functions successfully', async () => {
-        mockedSubscribers.onAnimationChanged(JSON.stringify(mockedAnimation));
+        await mockedSubscriberController.onAnimationChanged(JSON.stringify(mockedAnimation));
```
Comment (Matthiee) [inline src/tests/controllers/SubscriberContoller.test.ts:79]: The tests are grouped.

File: src/tests/controllers/AnimationController.test.ts [hunk 9]
```diff
@@ -1,70 +1,79 @@
...
+    jest.spyOn(mockEditorApi, 'setAnimationDuration');
+    jest.spyOn(mockEditorApi, 'resetFrameAnimation');
+    jest.spyOn(mockEditorApi, 'resetAnimation');
 });
 
 afterEach(() => {
     jest.restoreAllMocks();
 });
-describe('Animation methods', () => {
+describe('AnimationController', () => {
```
Comment (Matthiee) [inline src/tests/controllers/AnimationController.test.ts:42]: The tests are grouped.

**Review by Matthiee — CHANGES_REQUESTED** [review]
Looks good, I left some comments where the old "Should call all of the functions of child successfully" style was still used.

**Review by alexandraFlavia9 — APPROVED** [review]

**Review by Matthiee — APPROVED** [review]
LGTM, great work @brapoprod !

**Review by NicolaVerbeeck — APPROVED** [review]

**Review by alexandraFlavia9 — APPROVED** [review]

**Pull request merged**